cephfs/file.go

```diff
@@ -255,3 +255,39 @@ func (f *File) Fallocate(mode FallocFlags, offset, length int64) error {
 	ret := C.ceph_fallocate(f.mount.mount, f.fd, C.int(mode), C.int64_t(offset), C.int64_t(length))
 	return getError(ret)
 }
+
+// LockOp determines operations/type of locks which can be applied on a file.
+type LockOp int
+
+const (
+	// LockSH places a shared lock.
+	// More than one process may hold a shared lock for a given file at a given time.
+	LockSH = LockOp(C.LOCK_SH)
+	// LockEX places an exclusive lock.
+	// Only one process may hold an exclusive lock for a given file at a given time.
+	LockEX = LockOp(C.LOCK_EX)
+	// LockUN removes and existing lock held by this process.
+	LockUN = LockOp(C.LOCK_UN)
+	// LockNB can be ORed with any of the above to make a nonblocking call.
+	LockNB = LockOp(C.LOCK_NB)
+)
+
+// Flock applies or removes an advisory lock on an open file.
+//
+// Implements:
+//  int ceph_flock(struct ceph_mount_info *cmount, int fd, int operation, uint64_t owner);
+func (f *File) Flock(operation LockOp, owner uint64) error {
```

Comment:
The explicit owner thing here in the ceph call is a bit strange given that none of the other calls need you to pass an explicit value like this. But there's no Flock call in Go's os module so there's no conflict there. Maybe we could improve the documentation comment here to provide a little guidance about the parameters, but having them is just reflecting what ceph expects.

I had to stop and check exactly what owner is and ceph's doc comment says: "owner the user-supplied owner identifier (an arbitrary integer)". I think we should add something like this to our docs so people don't get too confused about what the values for this param should be. I almost suggested to defer the doc improvement to a later PR but this is likely more confusing than average. Plus no extra research is needed, just reflecting what ceph documents here will be good.

Reply:
Thanks for pointing it out, I observed the same while writing the tests.

Is this fine: 
// Flock applies or removes an advisory lock on an open file.
// Param owner is the user-supplied identifier for the owner of the
// lock, must be an integer.

Reply:
I think an important keyword to include is "arbitrary". It appears to just need to be a unique-enough cookie/token.

Reply:
Done, thanks. PTAL

Reply:
Seems reasonable, thanks.